extensions/stats/plugin.cc

```diff
@@ -192,46 +192,58 @@ const std::vector<MetricFactory>& PluginRootContext::defaultMetrics() {
       // HTTP, HTTP/2, and GRPC metrics
       MetricFactory{
           "requests_total", MetricType::Counter,
-
           [](const ::Wasm::Common::RequestInfo&) -> uint64_t { return 1; },
-          false},
+          false, count_standard_labels},
       MetricFactory{
           "request_duration_milliseconds", MetricType::Histogram,
           [](const ::Wasm::Common::RequestInfo& request_info) -> uint64_t {
             return request_info.duration /* in nanoseconds */ / 1000000;
           },
-          false},
+          false, count_standard_labels},
       MetricFactory{"request_bytes", MetricType::Histogram,
-
                     [](const ::Wasm::Common::RequestInfo& request_info)
                         -> uint64_t { return request_info.request_size; },
-                    false},
+                    false, count_standard_labels},
       MetricFactory{"response_bytes", MetricType::Histogram,
-
                     [](const ::Wasm::Common::RequestInfo& request_info)
                         -> uint64_t { return request_info.response_size; },
-                    false},
+                    false, count_standard_labels},
+      // GRPC streaming metrics.
+      // These metrics are dimensioned by peer labels as a minimum.
+      // TODO: consider adding connection security policy
+      MetricFactory{
+          "request_messages_count", MetricType::Counter,
+          [](const ::Wasm::Common::RequestInfo& request_info) -> uint64_t {
+            return request_info.request_message_count;
+          },
+          false, count_peer_labels},
+      MetricFactory{
+          "response_messages_count", MetricType::Counter,
+          [](const ::Wasm::Common::RequestInfo& request_info) -> uint64_t {
+            return request_info.response_message_count;
```

Comment:
how about metric for avg size of these messages?

Reply:
That would be difficult I think. The messages are framed by another filter, so we don't have a way to accumulate their sizes here. Open to any suggestions how to make that possible.

Reply:
I think it *would* be useful.
@kyessenov is it possible to add it upstream? We need a distribution of message sizes.
We need not  hold this PR for it though.

Reply:
We can report size distribution to native envoy stats, but I don't know how to get it out back here.

Reply:
not sure either... but can we use grpc stats filter only to get message size and add it in the above stats?

Reply:
There might be many messages, so we'd be passing a large list of values for sizes. We would need histogram summaries I think, to make it work.